libdnf/dnf-repo.c

```diff
@@ -1215,12 +1354,20 @@ dnf_repo_check_internal(DnfRepo *repo,
         if (!dnf_repo_set_timestamp_modified(repo, error))
             return FALSE;
         age_of_data =(g_get_real_time() - priv->timestamp_modified) / G_USEC_PER_SEC;
-        if (age_of_data > permissible_cache_age) {
+
+        /*choose a lower value between cache and metadata_expire for expired checking */
```

Comment:
So right now, `rpm-ostree compose tree` has a `--cache-only` option that does `dnf_context_set_cache_age (hifctx, G_MAXUINT);`.  I think yum does something similar conceptually when you pass `-C`.

Now, rpm-ostree could learn to iterate over all of the repos and call `dnf_repo_set_metadata_expire()`, but maybe we treat `permissible_cache_age == G_MAXUINT` as cache-only, and ignore the repo's `metadata_expire`?